File: dice-game/dice.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

// diceValues simulates rolling a specified number of dice with a given number of sides on each die.
func diceValues(numDice int, numSides int) int {
	sum := 0
	for d := 1; d <= numDice; d++ {
		// Initialize the random number generator with the current time's nanoseconds.
		rand.Seed(time.Now().UnixNano())
		// Generate a random value between 1 and numSides for each die.
		randomValue := rand.Intn(numSides) + 1
		fmt.Printf("Dice %d: %d\n", d, randomValue)
		sum += randomValue
	}
	return sum
}

func main() {
	numRolls := 2 // Number of times to roll the dice
	numDice := 3

	// The program simulates rolling the specified number of dice a specified number of times
	// and displays the results of each roll, including the total value rolled.
	for roll := 1; roll <= numRolls; roll++ {
		fmt.Println("Rolling:", roll)
		totalRoll := diceValues(numDice, 6) // Roll the dice and get the total
		fmt.Printf("The total rolled: %d\n", totalRoll)

		// Check for special cases based on the total rolled
		if totalRoll == 7 {
			fmt.Println("Lucky 7")
			if totalRoll%2 == 1 {
				fmt.Println("Odd")
			}
		} else if totalRoll == 2 && numDice == 2 {
			fmt.Println("Snake eyes (rolling two dice with a total of 2)")
		} else if totalRoll%2 == 0 {
			fmt.Println("Even")
		} else if totalRoll%2 == 1 {
			fmt.Println("Odd")
		}
	}
}

File: dice-game/dice.py
import random

def simulate_dice_rolls(numRolls, numDices, numSides):
    for roll_num in range(numRolls):
        print(f"Roll #{roll_num + 1}")
        
        # Simulate rolling multiple dice and calculate the total
        total_rolled = sum([random.randint(1, numSides) for _ in range(numDices)])
        print(f"The total rolled: {total_rolled}")
        
        # Check for special cases based on the total rolled
        if total_rolled == 7:
            print("Lucky 7")
        if total_rolled == 2 and numDices == 2:
            print("Snake Eyes")
        if total_rolled % 2 == 0:
            print("Even")
        if total_rolled % 2 ==1:
            print("Odd")

# Simulate 2 rolls of 3 six-sided dice
simulate_dice_rolls(2, 3, 6)
